amy/consents/migrations/0006_auto_20210614_1234.py

```diff
@@ -0,0 +1,148 @@
+# Generated by Django 2.2.17 on 2021-06-14 12:34
+
+from typing import Optional
+
+from django.db import migrations
+from django.utils import timezone
+
+AGREE = "agree"
+DECLINE = "decline"
+
+
+def reconsent(Consent, old_consent, term_option):
+    old_consent.archived_at = timezone.now()
+    old_consent.save()
+    Consent.objects.create(
+        term_id=term_option.term_id,
+        term_option=term_option,
+        person_id=old_consent.person_id,
+    )
+
+
+def copy_privacy_policy(apps, schema_editor):
+    Consent = apps.get_model("consents", "Consent")
+    TermOption = apps.get_model("consents", "TermOption")
+
+    term_slug = "privacy-policy"
+    try:
+        privacy_policy_agree = TermOption.objects.get(
+            term__slug=term_slug, archived_at=None
+        )
+    except TermOption.DoesNotExist:
+        return
+
+    for old_consent in Consent.objects.filter(
+        term__slug=term_slug, archived_at=None
+    ).select_related("person"):
+        if old_consent.person.data_privacy_agreement:
+            reconsent(Consent, old_consent, privacy_policy_agree)
+
+
+def copy_may_contact(apps, schema_editor):
+    Consent = apps.get_model("consents", "Consent")
+    TermOption = apps.get_model("consents", "TermOption")
+
+    term_slug = "may-contact"
+    try:
+        options = TermOption.objects.filter(term__slug=term_slug, archived_at=None)
+        may_contact_agree = [
+            option for option in options if option.option_type == AGREE
+        ][0]
+        may_contact_disagree = [
+            option for option in options if option.option_type == DECLINE
+        ][0]
+    except IndexError:
+        return
+
+    for old_consent in Consent.objects.filter(
+        term__slug=term_slug, archived_at=None
+    ).select_related("person"):
+        if old_consent.person.may_contact:
+            reconsent(Consent, old_consent, may_contact_agree)
+        else:
+            reconsent(Consent, old_consent, may_contact_disagree)
+
+
+def copy_public_profile(apps, schema_editor):
```

Comment:
The functions are very similar to each other. Perhaps in future you could try to use `partial` to reduce amount of code.

Reply:
That makes sense. Thanks for the tip.